### DISCUSSION TITLE
How to use templates for the projects?

### DISCUSSION THREAD

**Discussion by SiriusED** [premake/premake-core#2083] [Q&A] [answered]
Hi, I want something like this:
```
local project_template =
{

	{
		configurations  = { "Debug", "Release" },
		platforms       = { "x64", "x86" },
		defaultplatform = "x64",
		cdialect        = "C99",
		cppdialect      = "C++14",
		kind            = "ConsoleApp",
		filters         = { ... Filters here }
	}

}
```
And then use this object to generate actual projects files. But not generate lua files for each project, just use this template, generate some 'in-memory' string and then execute this string as script and generate files for the project. Using the template in some function in code, with some for loop to generate all projects in a single function, also acceptable for me.
Is it possible to do something like this with premake5?

**Comment by samsinsane** [comment]
Yeah, you can do whatever you want. It's just Lua with extra functions. `workspace "MyWorkspace"` is just a function, `workspace(config["workspace"])`, `_G[key](value)`.

**Reply by SiriusED** [reply]
> Yeah, you can do whatever you want. It's just Lua with extra functions. `workspace "MyWorkspace"` is just a function, `workspace(config["workspace"])`, `_G[key](value)`.

No, I know I can use it like lua table, but my question is rather - Can premake5 parse whole table structure described in lua table format and generate all visual studio files according this table data automatically?
It's like this: if I use valid field, for example `defaultplatform = "x64"` in the table then premake can parse it and use for generation, but if I use some invalid field it will give me an error that field is invalid and stop generation. Something like this.

**Reply by samsinsane** [reply, marked as answer]
I think you might have misunderstood my response, I was saying that you can write that functionality. Premake5 does not have a table format system, it uses functions.
